src/set.js

```diff
@@ -0,0 +1,81 @@
+class Set {
+  constructor(array = []){
+    this.array = array
+
```

Comment:
your set is an object with another object( an empty array) inside of it? and you add stuff to the array inside of it? My understanding is that the set actually holds the elements versus it having a property that holds elements...
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set